### PR TITLE
feat: Rewrite `uiSettings` mixin to a composable

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR will replace the use of the `uiSettings` mixin in the code base with the `useUiSettings` composable.

Fixes https://linear.app/chatwoot/issue/CW-3452/rewrite-uisettings-mixin-to-a-composable

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please take a look at the issue description.
https://linear.app/chatwoot/issue/CW-3452/rewrite-uisettings-mixin-to-a-composable#4e323935-4d12-4428-89ad-1fcac074f26e

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/profile/NotificationPreferences.vue [hunk 1]
```diff
@@ -242,31 +240,6 @@ export default {
...
-        this.updateUISettings({
-          alert_if_unread_assigned_conversation_exist: e.target.checked,
-        });
-      }
-      this.showAlert(this.$t('PROFILE_SETTINGS.FORM.API.UPDATE_SUCCESS'));
-    },
-    handleAudioToneChange(value) {
-      this.updateUISettings({ notification_tone: value });
-      this.showAlert(this.$t('PROFILE_SETTINGS.FORM.API.UPDATE_SUCCESS'));
-    },
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/profile/NotificationPreferences.vue:269]: These methods were removed as they are not used in this component.

File: app/javascript/dashboard/composables/useUISettings.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    if (!itemsOrder) {
+      return DEFAULT_CONVERSATION_SIDEBAR_ITEMS_ORDER;
+    }
+    // If the sidebar order doesn't have the new elements, then add them to the list.
+    DEFAULT_CONVERSATION_SIDEBAR_ITEMS_ORDER.forEach(item => {
+      if (!itemsOrder.find(i => i.name === item.name)) {
+        itemsOrder.push(item);
+      }
+    });
+    return itemsOrder;
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/composables/useUISettings.js]: Done https://github.com/chatwoot/chatwoot/pull/9819/commits/f2af1680edeed7fc0c028c6e4a43129d51c858a4

File: app/javascript/dashboard/composables/useUISettings.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  { name: 'conversation_info' },
+  { name: 'contact_attributes' },
+  { name: 'previous_conversation' },
+  { name: 'conversation_participants' },
+];
+export const DEFAULT_CONTACT_SIDEBAR_ITEMS_ORDER = [
+  { name: 'contact_attributes' },
+  { name: 'contact_labels' },
+  { name: 'previous_conversation' },
+];
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/composables/useUISettings.js]: Done https://github.com/chatwoot/chatwoot/pull/9819/commits/f2af1680edeed7fc0c028c6e4a43129d51c858a4

File: app/javascript/dashboard/composables/useUISettings.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+/**
+ * Slugifies the channel name.
+ * Replaces spaces, hyphens, and double colons with underscores.
+ * @param {string} name - The channel name to slugify.
+ * @returns {string} The slugified channel name.
+ */
+const slugifyChannel = name =>
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/composables/useUISettings.js:25]: @scmmishra I tried using it, but there is some difference in the results. 
For eg: `slugifyChannel` result = `channel_webwidget` `@sindresorhus/slugify` result = `channel_web_widget`

**Review by scmmishra — APPROVED** [review]

**Pull request merged**